github/plugin.py

```diff
@@ -88,3 +89,128 @@ def github_pulls(
         for obj in response_json
     ]
     return pull_requests
+
+
+@dataclass
+class GitHubIssue:
+    id: int
+    issue_number: int
+    issue_type: Literal["issue", "pull_request"]
+    html_url: str
+    created_at: datetime
+    updated_at: datetime
+    closed_at: Optional[datetime]
+    title: str
+    state: str
+    body: str
+    user_id: int
+    user_login: str
+    labels: list[str]
+
+
+def github_issues(
+    owner: str,
+    repo: str,
+    state: Literal["open", "closed", "all"] = "open",
+    sort: Literal["created", "updated", "comments"] = "created",
+    sort_direction: Literal["asc", "desc"] = "desc",
+    page: int = 1,
+) -> list[GitHubIssue]:
+    """
+    Returns results of a GitHub `issues` API call.
+
+    Issues can be either issues or pull requests. This function returns both.
```

Comment:
I wonder if the overlap with the explicit pull request call confuses things.  Maybe everything just works if we replace the `github_pulls` call?

Reply:
I am not sure, I'm happy to remove if there are no saved workflows using this yet.  The one difference is that github_pulls with page=1 will return pull requests whereas github_issues with page=1 may not if the first 30 issues are not PRs.  I think the callers should handle this but it is a difference.

Reply:
Yeah; maybe leave both for now given that they are exposed in the APIs themselves like that and do something later if we find that it is a problem.